src/java/uk/ac/rhul/cs/cl1/ui/cytoscape3/ExtractClusterAction.java

```diff
@@ -43,23 +46,37 @@ public void actionPerformed(ActionEvent event) {
 		}
 		
 		resultViewer.selectNodes(selectedNodes);
+		NewNetworkSelectedNodesAndEdgesTaskFactory taskFactory = 
+                                app.getService(CySwingAppAdapter.class).get_NewNetworkSelectedNodesAndEdgesTaskFactory();
 		
-		NewNetworkSelectedNodesAndEdgesTaskFatory taskFactory =
-				this.resultViewer.getCytoscapeApp().getService(NewNetworkSelectedNodesAndEdgesTaskFatory.class);
 		if (taskFactory == null) {
 			app.showBugMessage("Cannot create network representation for the cluster:\n" +
 					"New network creation factory is not registered.");
 			return;
 		}
 		
 		DialogTaskManager taskManager =
-				this.resultViewer.getCytoscapeApp().getService(DialogTaskManager.class);
+				app.getService(DialogTaskManager.class);
 		if (taskManager == null) {
 			app.showBugMessage("Cannot create network representation for the cluster:\n" +
 					"Dialog task manager is not registered.");
 			return;
 		}
 		
 		taskManager.execute(taskFactory.createTaskIterator(network));
+                this.resultViewer.incrementSubNetCount();
+                
+                
+                String currentNetworkName = network.getRow(network).get(CyNetwork.NAME, String.class);
+                Set<CyNetwork> allnetworks = app.getService(CyNetworkManager.class).getNetworkSet();
+                        
+                long maxSUID = Integer.MIN_VALUE;
+                for(CyNetwork net : allnetworks){
+                    if(net.getSUID() > maxSUID)
+                        maxSUID = net.getSUID();
+                }
+                CyNetwork newnet = app.getService(CyNetworkManager.class).getNetwork(maxSUID);
+                newnet.getRow(newnet).set(CyNetwork.NAME, currentNetworkName + " ClusterONE - SubNet " + this.resultViewer.getSubNetsExtracted());
+                
```

Comment:
Please fix the indentation of this code block; it is indented by one more space than the rest of the method.

Also, this block seems to rely on the fact that the most recently created network has the largest SUID. Is this guaranteed (i.e. documented) anywhere in the Cytoscape API?